src/modules/dashboard/sagas/actions.ts

```diff
@@ -1086,6 +1086,180 @@ function* editColonyAction({
+    }
+
+    if (!roles) {
+      throw new Error('Roles not set for setUserRole transaction');
+    }
+
+    txChannel = yield call(getTxChannel, metaId);
+
+    const batchKey = 'setUserRoles';
+
+    let annotationMessageIpfsHash = null;
+    if (annotationMessage) {
+      annotationMessageIpfsHash = yield call(
+        ipfsUpload,
+        JSON.stringify({
+          annotationMessage,
+        }),
+      );
+    }
+
+    const {
+      setUserRoles,
+      annotateSetUserRoles,
+    } = yield createTransactionChannels(metaId, [
+      'setUserRoles',
+      'annotateSetUserRoles',
+    ]);
+
+    const createGroupTransaction = ({ id, index }, config) =>
+      fork(createTransaction, id, {
+        ...config,
+        group: {
+          key: batchKey,
+          id: metaId,
+          index,
+        },
+      });
+
+    const roleArray = Object.values(roles).reverse();
+    roleArray.splice(2, 0, false);
+
+    let roleBitmask = '';
+
+    roleArray.forEach((role) => {
+      roleBitmask += role ? '1' : '0';
+    });
+
+    const hexString = hexlify(parseInt(roleBitmask, 2));
+    const zeroPadHexString = hexZeroPad(hexString, 32);
+
+    yield createGroupTransaction(setUserRoles, {
+      context: ClientType.ColonyClient,
+      methodName: 'setUserRolesWithProofs',
+      identifier: colonyAddress,
+      params: [userAddress, domainId, zeroPadHexString],
+      ready: false,
+    });
+
+    if (annotationMessage) {
+      yield createGroupTransaction(annotateSetUserRoles, {
+        context: ClientType.ColonyClient,
+        methodName: 'annotateTransaction',
+        identifier: colonyAddress,
+        params: [],
+        ready: false,
+      });
+    }
+
+    yield takeFrom(setUserRoles.channel, ActionTypes.TRANSACTION_CREATED);
+    if (annotationMessage) {
+      yield takeFrom(
+        annotateSetUserRoles.channel,
+        ActionTypes.TRANSACTION_CREATED,
+      );
+    }
+
+    yield put(transactionReady(setUserRoles.id));
+
+    const {
+      payload: { hash: txHash },
+    } = yield takeFrom(
+      setUserRoles.channel,
+      ActionTypes.TRANSACTION_HASH_RECEIVED,
+    );
+
+    yield takeFrom(setUserRoles.channel, ActionTypes.TRANSACTION_SUCCEEDED);
+
+    if (annotationMessage) {
+      yield put(
+        transactionAddParams(annotateSetUserRoles.id, [
+          txHash,
+          annotationMessageIpfsHash,
+        ]),
+      );
+
+      yield put(transactionReady(annotateSetUserRoles.id));
+
+      yield takeFrom(
+        annotateSetUserRoles.channel,
+        ActionTypes.TRANSACTION_SUCCEEDED,
+      );
+    }
+
+    yield put<AllActions>({
+      type: ActionTypes.COLONY_ACTION_USER_ROLES_SET_SUCCESS,
+      meta,
+    });
+
+    yield apolloClient.query<
+      ProcessedColonyQuery,
+      ProcessedColonyQueryVariables
+    >({
+      query: ProcessedColonyDocument,
+      variables: {
+        address: colonyAddress,
+      },
+      fetchPolicy: 'network-only',
+    });
```

Comment:
You don't need to run this at the end since no "actual" colony data needs to be refreshed

Reply:
I think you addressed this privately, as this still needs to be in.